rotkehlchen/chain/ethereum/modules/nft/nfts.py

```diff
@@ -280,36 +284,48 @@ def get_nfts_with_price(
             query_str += ' AND is_lp=?'
             bindings.append(lps_handling == NftLpHandling.ONLY_LPS)
 
+        if only_with_manual_prices is True:
+            query_str += ' AND manual_price=?'
+            bindings.append(1)
+
+        if from_asset is not None:
+            query_str += ' AND identifier=?'
+            bindings.append(from_asset.identifier)
+
+        if to_asset is not None:
+            query_str += ' AND last_price_asset=?'
+            bindings.append(to_asset.identifier)
+
         with self.db.conn.read_ctx() as cursor:
             query = cursor.execute(query_str, bindings)
             result = []
             for entry in query:
                 to_asset_id = entry[2] if entry[2] is not None else A_USD.identifier
                 try:
-                    to_asset = Asset(to_asset_id).check_existence()
+                    target_asset = Asset(to_asset_id).check_existence()
                 except UnknownAsset:
                     log.error(
                         f'Unknown asset {to_asset_id} in custom nft price DB table. Ignoring.',
                     )
                     continue
 
-                if to_asset != A_USD:
+                if target_asset != A_USD and only_with_manual_prices is False:
```

Comment:
So `if target_asset != A_USD and only_with_manual_prices is True:`

then `usd_price` is not really a usd price? Isn't that broken?

Reply:
I have changed it to make the field nullable and then the logic makes sense again. Also added the info to the docstring